### ISSUE TITLE
Forked mode sets unlocked account nonce to 0

### ISSUE THREAD

**Issue by chanhosuh** [nomiclabs/hardhat#1231]
Hardhat's forked mode sets unlocked accounts' nonces to 0, contrary to what they may be on the network being forked.

To replicate, use the test mnemonic `test test test test test test test test test test test junk` (the default for hardhat network in the config), which has [19 outgoing transactions according to Etherscan](https://etherscan.io/txs?a=0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266&f=2) for its account 0.

To check this, run the following script with hardhat network as forked mainnet.

```node
const { ethers} = require("hardhat");

async function main() {
  const signers = await ethers.getSigners();
  const deployer = signers[0];
  console.log("Deployer address:", deployer.address);
  const nonce = await ethers.provider.getTransactionCount(deployer.address);
  console.log("Deployer nonce:", nonce);  // 0, but expect 19
  console.log("");
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```

This can be considered a bug as it causes issues with contract deployment since a contract's address is fully determined by the sender's address and nonce.  Deploying a contract from an account with nonce 0 when it has already been used on the network being forked to deploy a contract at nonce 0 will cause a revert.

I believe a couple of the issues already reported here may be due to this:
- https://github.com/nomiclabs/hardhat/issues/1155
- https://github.com/nomiclabs/hardhat/issues/1204
- https://github.com/nomiclabs/hardhat/issues/1212
- https://github.com/nomiclabs/hardhat/issues/1233

**Comment by akhanubis** [comment]
I have been scratching my head for a few hours regarding reverted deploys. Have you found a workaround that doesnt involve using another address for the time being?

**Comment by fvictorio** [comment]
@akhanubis I haven't tested this, but an uglier workaround would be to send a lot of empty transactions from the first account and then start using it. That way there won't be collisions with the already deployed contracts from that address.

Right now the first account has a nonce of 19, so sending, say, 30 no-op txs should be more than enough.

**Comment by fvictorio** [comment]
Should be fixed in 2.0.10. Please re-open if it's not!
